process/block/poolsCleaner/miniBlocksPoolsCleaner_test.go

```diff
@@ -0,0 +1,132 @@
+package poolsCleaner
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewMiniBlocksPoolsCleaner_NilMiniblockPoolShouldErr(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, err := NewMiniBlocksPoolsCleaner(nil, &mock.RounderMock{}, &mock.CoordinatorStub{})
+
+	assert.Equal(t, process.ErrNilMiniBlockPool, err)
+	assert.Nil(t, miniblockCleaner)
+}
+
+func TestNewMiniBlocksPoolsCleaner_NilRounderShouldErr(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, err := NewMiniBlocksPoolsCleaner(&mock.CacherStub{}, nil, &mock.CoordinatorStub{})
+
+	assert.Equal(t, process.ErrNilRounder, err)
+	assert.Nil(t, miniblockCleaner)
+}
+
+func TestNewMiniBlocksPoolsCleaner_NilShardCoordinatorShouldErr(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, err := NewMiniBlocksPoolsCleaner(&mock.CacherStub{}, &mock.RoundStub{}, nil)
+
+	assert.Equal(t, process.ErrNilShardCoordinator, err)
+	assert.Nil(t, miniblockCleaner)
+}
+
+func TestNewMiniBlocksPoolsCleaner_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, err := NewMiniBlocksPoolsCleaner(&mock.CacherStub{}, &mock.RoundStub{}, &mock.CoordinatorStub{})
+
+	assert.Nil(t, err)
+	assert.NotNil(t, miniblockCleaner)
+}
+
+func TestReceivedMiniBlock_WrongTypeShouldBeIgnored(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, _ := NewMiniBlocksPoolsCleaner(&mock.CacherStub{}, &mock.RounderMock{}, &mock.CoordinatorStub{})
+
+	key := []byte("mbKey")
+	miniblock := &block.MiniBlockHeader{}
+	miniblockCleaner.receivedMiniBlock(key, miniblock)
+	assert.Nil(t, miniblockCleaner.mapMiniBlocksRounds[string(key)])
+}
+
+func TestReceivedMiniBlock_ShouldBeAddedInMap(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, _ := NewMiniBlocksPoolsCleaner(&mock.CacherStub{}, &mock.RounderMock{}, &mock.CoordinatorStub{})
+
+	key := []byte("mbKey")
+	miniblock := &block.MiniBlock{}
+	miniblockCleaner.receivedMiniBlock(key, miniblock)
+	assert.NotNil(t, miniblockCleaner.mapMiniBlocksRounds[string(key)])
+}
+
+func TestCleanMiniblocksPoolsIfNeeded_MiniblockNotInPoolShouldBeRemovedFromMap(t *testing.T) {
+	t.Parallel()
+
+	miniblockCleaner, _ := NewMiniBlocksPoolsCleaner(&mock.CacherStub{
+		GetCalled: func(key []byte) (value interface{}, ok bool) {
+			return nil, false
+		},
+	}, &mock.RounderMock{}, &mock.CoordinatorStub{})
```

Comment:
put each mock on a new line. Apply this in the entire PR, please

Reply:
done